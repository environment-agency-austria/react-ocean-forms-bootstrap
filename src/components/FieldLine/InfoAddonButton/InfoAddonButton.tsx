/**
 * Copyright (c) 2018-present, Umweltbundesamt GmbH
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React from 'react';

import { faExclamationCircle } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { Button, InputGroupAddon } from 'reactstrap';
import { useFormText } from 'react-ocean-forms';

import { IInfoAddonButtonProps } from './InfoAddonButton.types';

/**
 * Displays an input addon with an info icon button
 */
export const InfoAddonButton: React.FC<IInfoAddonButtonProps> = (props) => {
  const { info, plaintext, onClick } = props;

  const titleText = useFormText('ojs_show_information');

  if (plaintext) { return null; }
  if (info === undefined || info === '') { return null; }

  return (
    <InputGroupAddon addonType="append">
      <Button onClick={onClick} outline title={titleText}>
        <FontAwesomeIcon icon={faExclamationCircle} />
      </Button>
    </InputGroupAddon>
  );
};
