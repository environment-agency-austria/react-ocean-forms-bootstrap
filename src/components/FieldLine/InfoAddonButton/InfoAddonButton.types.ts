/**
 * Props for the InfoAddonButton component
 */
export interface IInfoAddonButtonProps {
  /**
   * Info text - will hide this button if empty
   */
  info?: string;
  /**
   * Form plaintext mode
   */
  plaintext: boolean;
  /**
   * OnClick event handler
   * @param event Event
   */
  onClick(event: React.MouseEvent): void;
}
