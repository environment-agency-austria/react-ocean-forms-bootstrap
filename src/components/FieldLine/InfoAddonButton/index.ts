export * from './InfoAddonButton';
export * from './InfoAddonButton.types';
