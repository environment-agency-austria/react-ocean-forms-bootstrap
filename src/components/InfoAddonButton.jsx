/**
 * Copyright (c) 2018-present, Umweltbundesamt GmbH
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React from 'react';
import PropTypes from 'prop-types';
import { Button, InputGroupAddon } from 'reactstrap';
import FontAwesomeIcon from '@fortawesome/react-fontawesome';
import faExclamationCircle from '@fortawesome/fontawesome-free-solid/faExclamationCircle';

/**
 * Displays an input addon with an info icon button
 */
function InfoAddonButton({ info, plaintext, onClick }) {
  if (plaintext) return null;
  if (!info) return null;

  return (
    <InputGroupAddon addonType="append">
      <Button onClick={onClick} outline>
        <FontAwesomeIcon icon={faExclamationCircle} />
      </Button>
    </InputGroupAddon>
  );
}

InfoAddonButton.defaultProps = {
  info: undefined,
};

InfoAddonButton.propTypes = {
  info: PropTypes.string,
  plaintext: PropTypes.bool.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default InfoAddonButton;
