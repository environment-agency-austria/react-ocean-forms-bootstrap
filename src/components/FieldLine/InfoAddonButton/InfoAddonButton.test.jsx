import React from 'react';
import { shallow } from 'enzyme';

import { InfoAddonButton } from './InfoAddonButton';

describe('<InfoAddonButton />', () => {
  const setup = (plaintext, info, onClick = () => {}) => shallow((
    <InfoAddonButton
      plaintext={plaintext}
      info={info}
      onClick={onClick}
    />
  ));

  it('should render nothing if meta.plaintext is active', () => {
    const wrapper = setup(true);
    expect(wrapper.exists('InputGroupAddon')).toBeFalsy();
  });

  it('should render nothing if there is no required field validator', () => {
    const wrapper = setup(false);
    expect(wrapper.exists('InputGroupAddon')).toBeFalsy();
  });

  describe('info present', () => {
    const MOCK_INFO = 'mock-info';
    const MOCK_CLICK_HANDLER = jest.fn();
    const wrapper = setup(false, MOCK_INFO, MOCK_CLICK_HANDLER);

    it('should render an InputGroupAddon', () => {
      expect(wrapper.exists('InputGroupAddon')).toBeTruthy();
    });

    it('should contain a Button with the click handler attached', () => {
      expect(wrapper.exists('Button')).toBeTruthy();
      expect(wrapper.find('Button').prop('onClick')).toBe(MOCK_CLICK_HANDLER);
    });

    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
