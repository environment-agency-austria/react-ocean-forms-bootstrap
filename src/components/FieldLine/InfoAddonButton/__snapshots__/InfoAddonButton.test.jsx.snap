// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfoAddonButton /> info present should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InfoAddonButton
    info="mock-info"
    onClick={[MockFunction]}
    plaintext={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "addonType": "append",
      "children": <Button
        color="secondary"
        onClick={[MockFunction]}
        outline={true}
        tag="button"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f06a",
                "M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z",
              ],
              "iconName": "exclamation-circle",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
      </Button>,
      "tag": "div",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f06a",
                "M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z",
              ],
              "iconName": "exclamation-circle",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />,
        "color": "secondary",
        "onClick": [MockFunction],
        "outline": true,
        "tag": "button",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "border": false,
          "className": "",
          "fixedWidth": false,
          "flip": null,
          "icon": Object {
            "icon": Array [
              512,
              512,
              Array [],
              "f06a",
              "M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z",
            ],
            "iconName": "exclamation-circle",
            "prefix": "fas",
          },
          "inverse": false,
          "listItem": false,
          "mask": null,
          "pull": null,
          "pulse": false,
          "rotation": null,
          "size": null,
          "spin": false,
          "symbol": false,
          "title": "",
          "transform": null,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "addonType": "append",
        "children": <Button
          color="secondary"
          onClick={[MockFunction]}
          outline={true}
          tag="button"
        >
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon={
              Object {
                "icon": Array [
                  512,
                  512,
                  Array [],
                  "f06a",
                  "M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z",
                ],
                "iconName": "exclamation-circle",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            title=""
            transform={null}
          />
        </Button>,
        "tag": "div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon={
              Object {
                "icon": Array [
                  512,
                  512,
                  Array [],
                  "f06a",
                  "M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z",
                ],
                "iconName": "exclamation-circle",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            title=""
            transform={null}
          />,
          "color": "secondary",
          "onClick": [MockFunction],
          "outline": true,
          "tag": "button",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "border": false,
            "className": "",
            "fixedWidth": false,
            "flip": null,
            "icon": Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f06a",
                "M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z",
              ],
              "iconName": "exclamation-circle",
              "prefix": "fas",
            },
            "inverse": false,
            "listItem": false,
            "mask": null,
            "pull": null,
            "pulse": false,
            "rotation": null,
            "size": null,
            "spin": false,
            "symbol": false,
            "title": "",
            "transform": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
